app/services/host_inventory_api.rb

```diff
@@ -7,16 +7,17 @@
 # Interact with the Insights Host Inventory. Usually HTTP calls
 # are all that's needed.
 class HostInventoryAPI
-  def initialize(host, account, url, b64_identity)
-    @host = host
+  def initialize(account, url)
     @url = "#{URI.parse(url)}#{ENV['PATH_PREFIX']}/inventory/v1/hosts"
     @account = account
-    @b64_identity = b64_identity
+    @b64_identity = Base64.strict_encode64(account.to_identity_header.to_json)
   end
 
-  def host_already_in_inventory
-    response = connection.get(@url, {}, 'X_RH_IDENTITY' => @b64_identity)
-    find_results(JSON.parse(response.body))
+  def hosts_already_in_inventory(hosts)
+    response = connection.get(
+      "#{@url}/#{hosts.map(&:id).join('%2C')}", {}, 'X_RH_IDENTITY' => @b64_identity
```

Comment:
To test, you may remove the `X_RH_IDENTITY` here , and override the `connection` with something like:

```ruby
  def connection
    Faraday.new do |f|
    faraday = Faraday.new do |f|
      f.response :raise_error
      f.adapter Faraday.default_adapter # this must be the last middleware
      f.ssl[:verify] = false
    end
    faraday.basic_auth 'user', 'pass'
    faraday
  end
end
```